### ISSUE TITLE
Logit Bias Error

### ISSUE THREAD

**Issue by ShelbyJenkins** [EricLBuehler/mistral.rs#398]
When adding logit_bias I hit this error. The same request works without logit bias.

```
thread '<unnamed>' panicked at /root/.cargo/git/checkouts/mistral.rs-0a2607fe9768eac5/00cfbab/mistralrs-core/src/pipeline/gguf.rs:645:9:
Expected receiver.: SendError { .. }
```

```
{"timestamp":"2024-06-05T23:19:36.712999Z","level":"INFO","error":"MistraRsBackend text_generation_request error: device mismatch in add, lhs: Cpu, rhs: Cuda { gpu_id: 0 }\n   0: candle_core::error::Error::bt\n             at /root/.cargo/git/checkouts/candle-c6a149c3b35a488f/5c06acd/candle-core/src/error.rs:231:25\n   1: candle_core::storage::Storage::same_device\n             at /root/.cargo/git/checkouts/candle-c6a149c3b35a488f/5c06acd/candle-core/src/storage.rs:60:17\n   2: candle_core::storage::Storage::binary_impl\n             at /root/.cargo/git/checkouts/candle-c6a149c3b35a488f/5c06acd/
```

looks to be here

```
/// Sample and add to the prefix cache.
#[macro_export]
macro_rules! do_sample {
    ($this:expr, $seqs:expr, $logits:expr, $prefix_cacher:expr, $disable_eos_stop:expr, $rng:expr) => {{
        let seqs_len = $seqs.len();
        let logits_seq = $logits.to_device(&Device::Cpu)?.chunk(seqs_len, 0)?;
        debug_assert_eq!(logits_seq.len(), seqs_len);
```

I think it's adding the logits to the Cpu? Not a blocker for me since I can use grammars. Thanks.

**Comment by EricLBuehler** [comment]
Hi @ShelbyJenkins, I should get a fix for this soon!

**Comment by EricLBuehler** [comment]
Hi @ShelbyJenkins! I just merged a fix for this in #454. Please feel free to reopen!
